wrangler-proto/src/main/java/io/cdap/wrangler/proto/workspace/v2/SampleSpec.java

```diff
@@ -29,13 +33,28 @@ public class SampleSpec {
   private final String connectionType;
   private final String path;
   private final Set<StageSpec> relatedPlugins;
+  // This gives the different methods the current connection supports for sampling
+  // (e.g. first N rows, randomized, stratified)
+  private final Set<SampleType> supportedSampleTypes;
+  private SampleRequest sampleRequest;
```

Comment:
Why do you need the request in the spec?

Reply:
Good question -- it's so that the frontend knows how each workspace was most recently sampled (it gets passed the SampleSpec object). I left a comment in the code to clarify.